src/hb-face.cc

```diff
@@ -96,11 +98,16 @@ DEFINE_NULL_INSTANCE (hb_face_t) =
 
 /**
  * hb_face_create_for_tables:
- * @reference_table_func: (closure user_data) (destroy destroy) (scope notified):
- * @user_data:
- * @destroy:
- *
+ * @reference_table_func: (closure user_data) (destroy destroy) (scope notified): Table-referencing function
+ * @user_data: A pointer to the user data
+ * @destroy: (optional): A callback to call when @data is not needed anymore
  *
+ * Variant of hb_face_create(), built for those cases where
+ * client programs are primarily concerned with querying the
+ * font tables, rather than with full shaping.
```

Comment:
I don’t think this is true. The faces created with `hb_face_create_for_tables()` can be used for any use (just like faces created with `hb_face_create()`). The difference is that instead of having a blob representing the full font data, a callback function is used to fetch the table data on demand. This can be used when interfacing with APIs that don’t give access to full font data (e.g. Core Text and Core Graphics APIs on macOS) or for whatever a reason the client finds this more appropriate than `hb_face_create()`.

Reply:
OK. Follow-up question for that case: is the callback function capable of providing access to everything that the blob could, or does it mean there are some restrictions?

Reply:
Generally there's no restriction in those method. Definitely not for shaping.

Currently our API for enumerating fine tables doesn't work with faces created this way.